relay-server/src/actors/envelopes.rs

```diff
@@ -1899,14 +1916,13 @@ impl EnvelopeManager {
         let future = rx
             .map_err(|_| SendEnvelopeError::SendFailed(UpstreamRequestError::ChannelClosed))
             .and_then(move |result| {
-                if let Err(UpstreamRequestError::RateLimited(upstream_limits)) = result {
-                    let limits = upstream_limits.scope(&scoping);
-                    ProjectCache::from_registry()
-                        .do_send(UpdateRateLimits::new(project_key, limits.clone()));
-                    Err(SendEnvelopeError::RateLimited(limits))
-                } else {
-                    result.map_err(SendEnvelopeError::SendFailed)
-                }
+                result.map_err(|err| {
+                    if let UpstreamRequestError::RateLimited(upstream_limits) = err {
+                        SendEnvelopeError::RateLimited(upstream_limits.scope(&scoping))
```

Comment:
SendEnvelopeError::RateLimited is only deconstructed once to produce a ProcessingError::RateLimited, which does not appear to ever touch those rate limits either. I believe you can remove the enum variant's field entirely (on both ProcessingError::RateLimited and SendEnvelopeError::RateLimited) and spare yourself the cloning and double-invocation to `scope()`.

Reply:
Yes, I'll do that